### PR TITLE
Only enable werror for debug

### LINKED ISSUE #5500: Compiling release fails on clang 16
There are lots of warnings emitted when using clang-16. We should consider disabling `Werror` for release builds to ensure they compile.

### PR DESCRIPTION
**Pull request by KingCol13**
Fixes #5500

### CODE REVIEW

File: SetFlags.cmake [hunk 1]
```diff
@@ -152,9 +152,6 @@ function(set_exe_flags TARGET)
 		target_compile_options(
 			${TARGET} PRIVATE
 
-			# Warnings-as-errors only on Clang for now:
-			-Werror
-
 			# TODO: actually fix the warnings instead of disabling them
```
Comment (12xx12) [inline SetFlags.cmake:155]: *cough*

**Comment by bearbin** [comment]
Not sure I agree with the premise of this change, why would we want to do less checking on release builds as compared to debug?

**Comment by KingCol13** [comment]
This is somewhat subjective, here's my reasoning:

- Compiler authors are free to implement arbitrarily stringent warnings
- We should treat release builds as being built my users
- Compilation by a user of a correct program should not fail just because they have a newer compiler with new warnings

Currently I think the easiest way to ensure the build completes with the current cmake is to pass in `CFLAGS="-Wno-everything" CXXFLAGS="-Wno-everything"`, turning off warnings entirely which isn't ideal.

Would you prefer adding a build flag, enabled by default to disable `Werror`, which we then disable in `compile.sh`?

I think we should make it as easy possible for new users compiling to get a working binary.

**Review by bearbin — APPROVED** [review]
Alright, this makes sense for now. Ideally we should be enabling errors in release mode as well during development (the code compiled in release mode is not necessarily the same as that compiled in debug); but this definitely does fix a pain point for users having brittle builds.

**Pull request merged**